src/rust/ide/src/view/node_editor.rs

```diff
@@ -435,17 +444,39 @@ impl GraphEditorIntegratedWithControllerModel {
 
 // === Handling Controller Notifications ===
 
-impl GraphEditorIntegratedWithControllerModel {
-    /// Handle notification received from controller.
-    pub fn handle_controller_notification
-    (&self, notification:&Option<controller::graph::executed::Notification>) {
-        use controller::graph::executed::Notification;
+impl GraphEditorIntegratedWithModelModel {
+    pub fn on_invalidated(&self) -> FallibleResult<()> {
+        self.refresh_graph_view()
+    }
+
+    pub fn on_node_stepped_into(&self, _id:double_representation::node::Id) -> FallibleResult<()> {
+        self.editor.frp.deselect_all_nodes.emit_event(&());
+        self.refresh_graph_view()
+    }
+
+    pub fn on_node_stepped_out(&self, id:double_representation::node::Id) -> FallibleResult<()> {
+        self.editor.frp.deselect_all_nodes.emit_event(&());
+        self.refresh_graph_view()?;
+        let id = self.get_displayed_node_id(id)?;
+        self.editor.frp.select_node.emit_event(&id);
+        Ok(())
+    }
+
+    pub fn on_values_computed(&self, expressions:&[ExpressionId]) -> FallibleResult<()> {
+        self.refresh_types_on(&expressions)
```

Comment:
missing docs. Please check why litner did not report it and remove allow missing docs which caused it!

Reply:
there's none, the type is not public